### PR TITLE
Make meter number nullable

### PR DESCRIPTION
**Pull request by erduna**
Meter number needs to be nullable, as it is not allowed for subtypes virtual or calculated

### CODE REVIEW

**Review by MrDach — DISMISSED** [review]

**Review by Korgath — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.Application/Create/CreateMeteringPointHandler.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ private static ProductionMeteringPoint CreateProductionMeteringPoint(CreateMeter
                 !string.IsNullOrEmpty(request.PowerPlant) ? GsrnNumber.Create(request.PowerPlant) : null,
                 request.LocationDescription,
                 EnumerationType.FromName<MeasurementUnitType>(request.UnitType),
-                request.MeterNumber,
+                request.MeterNumber!,
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.Application/Create/CreateMeteringPointHandler.cs:119]: ! ?

**Review by cmgrd — CHANGES_REQUESTED** [review]

File: source/Energinet.DataHub.MeteringPoints.Application/Create/CreateMeteringPointHandler.cs (hunk 1 above)
Comment (erduna) [reply source/Energinet.DataHub.MeteringPoints.Application/Create/CreateMeteringPointHandler.cs:119]: It's nullable, thus the !. The other nullables do the same thing

**Review by Korgath — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.Application/Create/CreateMeteringPointHandler.cs (hunk 1 above)
Comment (Sondergaard) [reply source/Energinet.DataHub.MeteringPoints.Application/Create/CreateMeteringPointHandler.cs]: This just seems odd. 

With the ! you say I know this is not null. I don't see any statements that checks this assumption. 

If MeterNumber is really nullable as the signature indicates, then the API that receives the value should also accept it as nullable.

Comment (erduna) [reply source/Energinet.DataHub.MeteringPoints.Application/Create/CreateMeteringPointHandler.cs]: It's fixed in the subsequent commits, we had a talk offline about the issue :)

File: source/Energinet.DataHub.MeteringPoints.Application/Validation/Rules/MeterNumberMustBeValidRule.cs [hunk 2]
```diff
@@ -34,7 +34,7 @@ public MeterNumberMustBeValidRule()
 
                 RuleFor(createMeteringPoint => createMeteringPoint.MeterNumber)
                     .MaximumLength(MeterNumberMaximumLength)
-                    .WithState(createMeteringPoint => new MeterNumberMaximumLengthValidationError(createMeteringPoint.GsrnNumber, createMeteringPoint.MeterNumber, MeterNumberMaximumLength));
+                    .WithState(createMeteringPoint => new MeterNumberMaximumLengthValidationError(createMeteringPoint.GsrnNumber, createMeteringPoint.MeterNumber!, MeterNumberMaximumLength));
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.Application/Validation/Rules/MeterNumberMustBeValidRule.cs:37]: Should there be a bang here?

Comment (erduna) [reply source/Energinet.DataHub.MeteringPoints.Application/Validation/Rules/MeterNumberMustBeValidRule.cs:36]: Yes. Fixed other comments in latest push :)

**Review by cmgrd — APPROVED** [review]

**Review by Korgath — APPROVED** [review]

**Pull request merged**